### PR TITLE
Ramping fixes for Echo and Reverb effects

### PR DESCRIPTION
**Pull request by Be-ing**
This continues from #1703 and #1676 to fix the pop sound when enabling/disabling the Echo and Reverb effects in a D+W chain with no other effects in the chain enabled. That was caused by the inappropriate ramping in EngineEffect::process.

### CODE REVIEW

**Comment by Be-ing** [comment]
@daschuer in #1703:
> In  the manifest, we have the fade to dry flag, which tells the effect  engine if the effect already has internal fade. If you remove the dry  bypass, it can't probably work as intended. So I think both effects need  to handle the ramping in this case as well and also switching of the  Mixxx mode.

You are correct. The ramping done by `EngineEffect::process` was inappropriately applying a ramped gain to the input buffer when enabling/disabling Echo and Reverb. Upon closer inspection, I realized that Echo and Reverb should have been handling their own ramping on/off all along. `EngineEffect::process`'s ramping on enabling doesn't make sense for these effects because they don't start outputting sound immediately. Instead it is up to the effects to ramp their Send parameters from 0 when enabling. I also realized that Reverb wasn't ramping its Send parameter at all.

File: src/util/rampingvalue.h [hunk 1]
```diff
@@ -1,22 +1,58 @@
...
+// used when iterating over the audio buffer, which is typically the number of
+// frames or samples in the audio engine's buffer. At the start of each audio
+// engine cycle, before the parameter is used, call
+// RampingValue::setCurrentCallbackValue with the parameter value received from
+// outside the audio engine thread. Then, when looping over the audio buffer,
+// get the interpolated value with RampingValue::rampedValue
 template <typename T>
 class RampingValue {
   public:
-    RampingValue(const T& initial, const T& final, int steps) {
```
Comment (daschuer) [inline src/util/rampingvalue.h]: Sorry to say, but my old implementation was way better, because it was designed to live on the fast stack.
Originally it requires only the old value to be in the state class, now we have four values that have to be fetched from the heap each call. The original getNext function only did a =+ return on the fast stack. Now we have four lines accessing heap memory. So please revert the changes.

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/util/rampingvalue.h (hunk 1 above)
Comment (Be-ing) [reply src/util/rampingvalue.h]: Okay, it will be easier to fix the bugs with Echo and Reverb starting in a new branch, so I'll close this PR.
